pkg/cmd/join/cmd.go

```diff
@@ -47,5 +47,6 @@ func NewCmd(clusteradmFlags *genericclioptionsclusteradm.ClusteradmFlags, stream
 	cmd.Flags().StringVar(&o.hubAPIServer, "hub-apiserver", "", "The api server url to the hub")
 	cmd.Flags().StringVar(&o.clusterName, "cluster-name", "", "The name of the joining cluster")
 	cmd.Flags().StringVar(&o.outputFile, "output-file", "", "The generated resources will be copied in the specified file")
+	cmd.Flags().StringVar(&o.timeout, "timeout", "", "extend timeout from 300 secounds ")
```

Comment:
I think you could put it as a global flags here
https://github.com/open-cluster-management-io/clusteradm/blob/main/pkg/genericclioptions/clusteradm_flags.go#L23

and use IntVar so you would not need to convert from string to int.